app/lib/features/events/pages/event_list_page.dart

```diff
@@ -0,0 +1,224 @@
+            : null,
+        onChanged: (value) =>
+            ref.read(searchValueProvider.notifier).state = value,
+      ),
+    );
+  }
+
+  Widget filterChipsButtons() {
+    return SingleChildScrollView(
+      scrollDirection: Axis.horizontal,
+      child: Container(
+        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
+        child: Row(
+          children: [
+            FilterChip(
+              selected: eventFilterValue == EventFilters.all,
+              label: Text(L10n.of(context).all),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.all,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.ongoing,
+              label: Text(L10n.of(context).ongoing),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.ongoing,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.upcoming,
+              label: Text(L10n.of(context).upcoming),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.upcoming,
+            ),
+            const SizedBox(width: 10),
+            FilterChip(
+              selected: eventFilterValue == EventFilters.past,
+              label: Text(L10n.of(context).past),
+              onSelected: (value) => ref
+                  .read(eventFilerProvider.notifier)
+                  .state = EventFilters.past,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildEventList(List<CalendarEvent> events) {
+    final size = MediaQuery.of(context).size;
+    final widthCount = (size.width ~/ 500).toInt();
+    const int minCount = 2;
+
+    if (events.isEmpty) return _buildEventsEmptyState();
+
+    return SingleChildScrollView(
+      child: StaggeredGrid.count(
+        crossAxisCount: max(1, min(widthCount, minCount)),
+        children: [
+          for (var event in events)
+            EventItem(
+              event: event,
+              isShowSpaceName: widget.spaceId == null,
+            ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildEventsEmptyState() {
+    bool canAdd = false;
+    if (searchValue.isEmpty) {
+      canAdd = ref
+              .watch(hasSpaceWithPermissionProvider('CanPostEvent'))
+              .valueOrNull ??
+          false;
+    }
+    return Center(
+      heightFactor: 1,
+      child: EmptyState(
+        title: searchValue.isNotEmpty
+            ? L10n.of(context).noMatchingEventsFound
+            : L10n.of(context).noEventsFound,
+        subtitle: L10n.of(context).noEventAvailableDescription,
+        image: 'assets/images/empty_event.svg',
+        primaryButton: canAdd && searchValue.isEmpty
```

Comment:
you are checking `searchValue` unnecessarily twice.
```suggestion
        primaryButton: canAdd
```